code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -80,220 +123,283 @@
+		user.drop_item()
 
-			icon_state = "broken"
-			status = newstatus
+		forceMove(get_step(newtarget.loc, reverse_direction(direction)))
+		set_dir(reverse_direction(direction))
+		status = STATUS_ACTIVE
+		attach(newtarget)
+		return
 
 /obj/item/device/magnetic_lock/proc/setconstructionstate(var/newstate)
-	constructionstate = newstate
-	if (newstate == 0)
-		if (status == STATUS_ACTIVE)
-			icon_state = "active"
-		else
-			icon_state = "inactive"
-	else if (newstate == 2)
-		flick("deconstruct_2_anim", src)
+	if (!powercell && newstate == 1)
+		setconstructionstate(2)
+		return
 	else if (newstate == 4)
-		setstatus(STATUS_BROKEN)
-	else
-		icon_state = "deconstruct_[constructionstate]"
+		detach(playflick = 0)
+		update_icon()
+	constructionstate = newstate
+	update_icon()
 
 /obj/item/device/magnetic_lock/proc/detach(var/playflick = 1)
 	if (target)
 
 		if (playflick)
-			spawn(-15) flick("release", src)
+			spawn(-15) flick("release_[department]", src)
 
-		adjustsprite(null)
+		status = STATUS_INACTIVE
+		set_dir(SOUTH)
+		update_icon()
 		layer = LAYER_NORMAL
 
 		target.bracer = null
+		target = null
+		if (target_node1)
+			target_node1.bracer = null
+			target_node1 = null
+		if (target_node2)
+			target_node2.bracer = null
+			target_node2 = null
+		anchored = 0
 
 		processing_objects.Remove(src)
-
-		anchored = 0
+		last_process_time = 0
 
 /obj/item/device/magnetic_lock/proc/attach(var/obj/machinery/door/airlock/newtarget as obj)
-	adjustsprite(newtarget)
 	layer = LAYER_ATTACHED
-	flick("deploy", src)
 
 	newtarget.bracer = src
 	target = newtarget
 
+	last_process_time = world.time
 	processing_objects.Add(src)
-
 	anchored = 1
 
-/obj/item/device/magnetic_lock/proc/adjustsprite(var/obj/target as obj)
-	if (target)
-		switch (get_dir(src, target))
+	spawn(-15)
+		flick("deploy_[department]", src)
+	update_icon()
+
+/obj/item/device/magnetic_lock/update_icon()
+	if (status == STATUS_ACTIVE && target)
+		icon_state = "active_[department]"
+		switch (dir)
 			if (NORTH)
 				pixel_x = 0
-				pixel_y = 32
-			if (NORTHEAST)
-				pixel_x = 32
-				pixel_y = 32
+				pixel_y = -32
 			if (EAST)
-				pixel_x = 32
+				pixel_x = -32
 				pixel_y = 0
-			if (SOUTHEAST)
-				pixel_x = 32
-				pixel_y = -32
 			if (SOUTH)
 				pixel_x = 0
-				pixel_y = -32
-			if (SOUTHWEST)
-				pixel_x = -32
-				pixel_y = -32
+				pixel_y = 32
 			if (WEST)
-				pixel_x = -32
+				pixel_x = 32
 				pixel_y = 0
-			if (NORTHWEST)
-				pixel_x = -32
-				pixel_y = 32
+	else if (status >= STATUS_INACTIVE)
+		icon_state = "inactive_[department]"
+		pixel_x = 0
+		pixel_y = 0
 	else
+		icon_state = "broken_[department]"
 		pixel_x = 0
 		pixel_y = 0
+	update_overlays()
+
+/obj/item/device/magnetic_lock/proc/update_overlays()
+	overlays.Cut()
+	switch (status)
+		if (STATUS_BROKEN)
+			icon_state = "broken"
+			return
+
+		if (STATUS_INACTIVE to STATUS_ACTIVE)
+			if (hacked)
+				overlays += "overlay_hacked"
```

Comment:
See note about overlays.

This shouldn't work? Unless it does. At which point, let me know, so I know that I've been lied to all this time.